src/datamodel/validator/Uuid.cpp

```diff
@@ -10,3 +10,139 @@
 // SPDX-License-Identifier: Apache-2.0
 
 #include "up-cpp/datamodel/validator/Uuid.h"
+
+#include <uprotocol/v1/uuid.pb.h>
+
+#include <chrono>
+#include <optional>
+#include <stdexcept>
+#include <tuple>
+
+namespace uprotocol::datamodel::validator::uuid {
+std::string_view message(Reason reason) {
+	switch (reason) {
+		case Reason::WRONG_VERSION:
+			return "The UUID could not be interpreted as a uP v8 UUID";
+		case Reason::UNSUPPORTED_VARIANT:
+			return "The UUID variant is not a supported variant";
+		case Reason::FROM_THE_FUTURE:
+			return "The time in the UUID is in the future";
+		case Reason::EXPIRED:
+			return "The UUID has expired";
+		default:
+			return "Unknown reason";
+	}
+}
+
+using ValidationResult = std::tuple<bool, std::optional<Reason>>;
+
+ValidationResult isUuid(const uprotocol::v1::UUID uuid) {
+	uint64_t msb = uuid.msb();
+	uint64_t lsb = uuid.lsb();
+
+	uint8_t version = (msb >> 12) & 0xF;
+	if (version != 8) {
+		return {false, Reason::WRONG_VERSION};
+	}
+
+	uint8_t variant = (lsb >> 62) & 0x3;
+	if (variant != 2) {  // Variant should be 10 in binary
+		return {false, Reason::UNSUPPORTED_VARIANT};
+	}
+
+	uint64_t timestamp =
+	    (msb >> 16) & 0xFFFFFFFFFFFF;  // Extract 48-bit timestamp
+	uint64_t current_time =
+	    std::chrono::duration_cast<std::chrono::milliseconds>(
+	        std::chrono::system_clock::now().time_since_epoch())
+	        .count();
```

Comment:
I would generally recommend against using `.count()` in this context. It would be safer to convert the timestamp from the UUID to a time point:
```suggestion
	std::chrono::milliseconds uuid_time(
	    (msb >> 16) & 0xFFFFFFFFFFFF);  // Extract 48-bit timestamp
	std::chrono::system_clock::time_point timestamp(uuid_time);
	auto current_time = std::chrono::system_clock::now();
```

Reply:
It might also make sense to put a `getUuidTimestamp()` method in the anonymous namespace of this file so the code doesn't need to be repeated for each function